examples/snake/client/src/services/ClearNetService.ts

```diff
@@ -337,16 +347,18 @@ class ClearNetService {
     }
 
     async getActiveChannel(): Promise<Hex | null> {
-        if (this.activeChannel) {
-            return this.activeChannel;
-        }
-
-        const channels = await this.client.getAccountChannels();
-        if (channels.length <= 0) {
-            throw new Error('No active channel found. Please open a channel at apps.yellow.com');
+        // Wait until broker pushes the active channel to the client
+        let attempts = 0;
+        const timeout = 100;
+        const maxAttempts = 2000 / timeout;
+        while (!this.activeChannel) {
+            await new Promise(resolve => setTimeout(resolve, timeout));
+            attempts++;
+            if (this.isConnected && attempts > maxAttempts) {
+                throw new Error('No active channel found. Please open a channel at apps.yellow.com');
+            }
         }
-        this.activeChannel = channels[0];
-
+        console.log('[ClearNetService] Active channel:', this.activeChannel);
         return this.activeChannel;
```

Comment:
You don't need to wait. If it returns an empty list in channels (welcome message event), it means you don't have any to verify. u can also use `get_channels` by participant